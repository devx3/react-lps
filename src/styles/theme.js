export const theme = {
    colors: {
        primaryColor: '#0A1128',
        secondaryColor: '#dc143c',
        white: '#FFFFFF',
        mediumGrey: '#DDDDDD',
    },
    fonts: {
        family: {
            default: "'Open Sans', sans-serif;",
            secondary: "'Montserrat', sans-serif;",
        },
        sizes: {
            smallest: '0.8rem',
            xsmall: '1.2rem',
            small: '1.6rem',
            medium: '2.4rem',
            big: '3.2rem',
            biggest: '4.0rem',
            extraBig: '4.8rem',
            huge: '5.6rem',
            extraHuge: '6.4rem',
        },
    },
    media: {
        ltMedia: '(max-width: 768px)',
    },
    spacings: {
        xsmall: '8rem',
        small: '1.6rem',
        medium: '2.4rem',
        large: '3.2rem',
        xlarge: '4.0rem',
        xxlarge: '4.8rem',
        huge: '5.6rem',
        xhuge: '6.4rem',
    },
};
