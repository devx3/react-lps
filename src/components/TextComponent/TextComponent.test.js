import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme.js';
import { theme } from '../../styles/theme.js';
import { TextComponent } from './index.jsx';

describe('<TextComponent />', () => {
    it('Should render a text', () => {
        renderTheme(<TextComponent>Lorem ipsaum</TextComponent>);
        expect(screen.getByText('Lorem ipsaum')).toBeInTheDocument();
    });

    it('Should match with snapshot', () => {
        renderTheme(<TextComponent>Lorem ipsaum</TextComponent>);
        expect(screen.getByText('Lorem ipsaum')).toMatchSnapshot();
    });

    it('Should render with the small font', () => {
        renderTheme(<TextComponent>Lorem ipsaum</TextComponent>);
        expect(screen.getByText('Lorem ipsaum')).toHaveStyleRule(
            'font-size',
            theme.fonts.sizes.small,
        );
    });
});
