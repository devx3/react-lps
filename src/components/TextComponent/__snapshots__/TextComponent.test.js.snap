// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextComponent /> Should match with snapshot 1`] = `
.c0 {
  font-size: 1.6rem;
}

<div
  class="c0"
>
  Lorem ipsaum
</div>
`;
